Deprecate withEnterpriseEdition due to version pinning

meistermeier opened the pull request:
Deprecate the method in favour of a version-agnostic replacement. The `withEnterpriseEdition` pulls in the former neo4j:4.4 default image / LTS version without users noting the implicit change from their defined image version.
A `Neo4jContainer("neo4j:5.26").withEnterpriseEdition()` becomes `neo4j:4.4-enterprise` and let's say that this is at least a surprise, even though it's explicitly mentioned in the documentation.

Also upgrade the test images to be used to 5.26 (latest LTS version) where possible.

eddumelendez commented on modules/neo4j/src/main/java/org/testcontainers/containers/Neo4jContainer.java:
```diff
@@ -211,8 +206,7 @@ public Neo4jContainer withEnterpriseEdition() {
         }
-
-        setDockerImageName(DEFAULT_IMAGE_NAME.withTag(ENTERPRISE_TAG).asCanonicalNameString());
+        setDockerImageName(DockerImageName.parse(getDockerImageName() + ENTERPRISE_SUFFIX).asCanonicalNameString());
```
this will break existing users. Why not improve the exception's message?

meistermeier replied:
Will play with this. Thanks for the feedback.

meistermeier commented:
Updated the PR do work with main after the JUnit upgrade. 
I also revisited the comment above and looked at the documentation where we clearly state that it's only for 4.4. As a consequence I reverted this portion in the PR.

kiview commented:
@meistermeier I'm probably missing something since I was out of the game for some time, but isn't the only thing this PR effectively does now, is bumping the image in the tests (without effective changes to the module behavior besides an internal refactoring)? 

Or asked differently, which usage was failing before, that would work now? (which we could add as a new test accordingly)

meistermeier commented:
You're pretty much spot-on now... The test (and minor renaming) is all that's left. But it's an improvement for the test suite nevertheless because it targets (mostly) everywhere the latest LTS. But for now it's the only thing I can add if the breaking API with a real improvement does not get accepted.

meistermeier commented:
Naming is so hard, so I came up with `withEnterpriseImage` for now. I am happy for better suggestions. Just tried to keep the `with...` pattern but 🤷 
There is no reason for me to use something like `withEnterpriseEdition(someparam)` because:
* The tag was already enforced by the constructor
* The acceptance is done by the acceptance file

kiview reviewed (dismissed):
@meistermeier I think that is a nice _hack_ to deprecate the problematic old method and add the new desired behavior, _without_ ~breaking existing users.

LGTM, any objections @eddumelendez ?

kiview commented on modules/neo4j/src/main/java/org/testcontainers/containers/Neo4jContainer.java:
```diff
@@ -204,15 +205,40 @@ public String getHttpsUrl() {
+     * Community Edition can be used for free in your projects under the GPL v3 license, Neo4j Enterprise edition
+     * needs either a commercial, education or evaluation license.
+     *
+     * @return This container
+     **/
```
Take it or leave it 🤷

meistermeier commented on modules/neo4j/src/test/java/org/testcontainers/containers/Neo4jContainerTest.java:
```diff
@@ -81,7 +83,7 @@ void shouldFailOnCopyDatabaseForDefaultNeo4j4Image() {
         assertThatIllegalArgumentException()
             .isThrownBy(() -> {
-                new Neo4jContainer("neo4j:4.4.1").withDatabase(MountableFile.forClasspathResource("/test-graph.db"));
+                new Neo4jContainer("neo4j:5.26.10").withDatabase(MountableFile.forClasspathResource("/test-graph.db"));
```
Yes, I want to be safe that the 4, 5, 6(latest) versions are correctly parsed to fail with the expected message.

meistermeier commented:
Thanks for the review @kiview, PR updated.

meistermeier commented:
Closing this PR because it's no somehow outdated and it was not included in the 2.x line where a breaking change might be much easier to embed.
